Src/WitsmlExplorer.Api/HttpHandlers/EssentialHeaders.cs

```diff
@@ -1,50 +1,43 @@
-
-using System;
-
 using Microsoft.AspNetCore.Http;
-
 namespace WitsmlExplorer.Api.HttpHandlers
 {
     public interface IEssentialHeaders
     {
+
         public string Authorization { get; }
-        public bool HasCookieCredentials(string server);
-        public string GetCookie(string server);
-        public string GetHost(string server);
+        public string GetCookieValue();
+        public string GetHeaderValue(string witsmlServer);
         public string GetBearerToken();
     }
 
     public class EssentialHeaders : IEssentialHeaders
     {
+        public static readonly string CookieName = "witsmlexplorer";
         public static readonly string WitsmlTargetServer = "WitsmlTargetServer";
         public static readonly string WitsmlSourceServer = "WitsmlSourceServer";
+
         public EssentialHeaders() { }
-        public EssentialHeaders(HttpRequest httpRequest)
+        public EssentialHeaders(HttpRequest httpRequest, string environment)
         {
 
             Authorization = httpRequest?.Headers["Authorization"];
             TargetServer = httpRequest?.Headers[WitsmlTargetServer];
             SourceServer = httpRequest?.Headers[WitsmlSourceServer];
-            TargetServerCookie = httpRequest?.Cookies[Uri.EscapeDataString(TargetServer)];
-            SourceServerCookie = httpRequest?.Cookies[Uri.EscapeDataString(SourceServer)];
+            WitsmlExplorerCookie = httpRequest?.Cookies[$"{CookieName}-{environment}"];
         }
         public string Authorization { get; init; }
         private string TargetServer { get; init; }
         private string SourceServer { get; init; }
-        private string TargetServerCookie { get; init; }
-        private string SourceServerCookie { get; init; }
+        private string WitsmlExplorerCookie { get; init; }
 
-        public bool HasCookieCredentials(string server)
-        {
-            return (server == WitsmlTargetServer) ? !string.IsNullOrEmpty(TargetServerCookie) : !string.IsNullOrEmpty(SourceServerCookie);
-        }
-        public string GetCookie(string server)
+
+        public string GetCookieValue()
         {
-            return (server == WitsmlTargetServer) ? TargetServerCookie : SourceServerCookie;
+            return WitsmlExplorerCookie;
         }
-        public string GetHost(string server)
+        public string GetHeaderValue(string witsmlServer)
```

Comment:
Would it be better to make TargetServer and SourceServer public to make retrieving them cleaner that calling GetHeaderValue with the static strings?

Reply:
Yes I think so too, will change that

Reply:
Also I feel that the name GetHeaderValue does not imply that it can be only used to retrieve either the TargetServer or the SourceServer header.

Reply:
We does not consider any other headers to be `essential` (except for the Authorization ofcourse) so this was the reasoning behind.
In this case using GetHeaderValue ended up being a bit overkill so I will remove that